packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -95,17 +95,39 @@ hub.post("/", async (req, res) => {
       // Check if bot is running on a non-default chain, and fetch last block number seen on this or the default chain.
       const [botWeb3, spokeCustomNodeUrl] = _getWeb3AndUrlForBot(configObject[botName]);
       const chainId = await _getChainId(botWeb3);
-      nodeUrlToChainIdCache[spokeCustomNodeUrl] = chainId;
-
-      // If we've seen this chain ID already we can skip it:
+      // If we've seen this chain ID already we can skip it.
       if (blockNumbersForChain[chainId]) continue;
 
-      // Fetch last seen block for this chain:
-      let lastQueriedBlockNumber = await _getLastQueriedBlockNumber(req.body.configFile, chainId, logger);
+      nodeUrlToChainIdCache[spokeCustomNodeUrl] = chainId;
+
+      // If STORE_MULTI_CHAIN_BLOCK_NUMBERS is set then this bot requires to know a number of last seen blocks across
+      // a set of chainIds. Construct a batch promise to evaluate the latest block number for each chainId.
+      if (configObject[botName]?.environmentVariables?.STORE_MULTI_CHAIN_BLOCK_NUMBERS) {
+        const multiChainIds = configObject[botName]?.environmentVariables?.STORE_MULTI_CHAIN_BLOCK_NUMBERS;
+        let promises = [];
+        for (const chainId of multiChainIds) {
+          promises.push(
+            _getLastQueriedBlockNumber(req.body.configFile, chainId, logger),
+            _getBlockNumberOnChainIdMultiChain(configObject[botName], chainId)
+          );
+        }
+        let results = await Promise.all(promises);
+        results.forEach((_, index) => {
+          if (index % 2 !== 0) return;
+          const chainId = multiChainIds[Math.floor(index / 2)];
```

Comment:
nit: why is math floor needed here? Shouldn't these always be even?

Reply:
o ye, you right. the line above actually protects this from not being the case.